### PR TITLE
Added --skip-login flag to docker-pull and docker-push.

### PR DESCRIPTION
**Pull request by eyalbe4**

### CODE REVIEW

**Comment by djgilcrease** [comment]
This looks like a better option then implicitly skipping as my PR does https://github.com/jfrog/jfrog-cli-go/pull/353 feel free to close the other PR

File: artifactory/cli.go [hunk 1]
```diff
@@ -680,6 +680,13 @@ func getBuildToolFlags() []cli.Flag {
 	}
 }
 
+func getSkipLoginFlag() cli.Flag {
```
Comment (AlexeiVainshtein) [inline artifactory/cli.go]: Both docker push and docker pull are using the same flags such as:
getBuildToolFlags, getServerFlags and getSkipLoginFlag. The only difference is in the docker push command that has also the getThreadsFlag. Perhaps we should create a getDockerFlags that will include the getBuildToolFlags, getServerFlags and getSkipLoginFlag and for docker push also add the threads flags:

```
func getDockerFlags() []cli.Flag {
	var flags []cli.Flag
	flags = append(flags, getBuildToolFlags()...)
	flags = append(flags, getServerFlags()...)
	flags = append(flags, getSkipLoginFlag())
	return flags
}

func getDockerPushFlags() []cli.Flag {
	var flags []cli.Flag
	flags = append(flags, getDockerFlags()...)
	flags = append(flags, getThreadsFlag())
	return flags
}
```

File: artifactory/cli.go [hunk 2]
```diff
@@ -1324,10 +1333,12 @@ func dockerPushCmd(c *cli.Context) {
 	artDetails := createArtifactoryDetailsByFlags(c, true)
 	imageTag := c.Args().Get(0)
 	targetRepo := c.Args().Get(1)
+		skipLogin := c.Bool("skip-login")
+
```
Comment (AlexeiVainshtein) [inline artifactory/cli.go]: Redundant white spaces

File: artifactory/commands/docker/pull.go [hunk 3]
```diff
@@ -24,10 +24,12 @@ func (dpc *DockerPullCommand) Run() error {
 	if errorutils.CheckError(err) != nil {
 		return err
 	}
-	loginConfig := &docker.DockerLoginConfig{ArtifactoryDetails: rtDetails}
-	err = docker.DockerLogin(imageTag, loginConfig)
-	if err != nil {
-		return err
+	if !dpc.skipLogin {
+		loginConfig := &docker.DockerLoginConfig{ArtifactoryDetails: rtDetails}
+		err = docker.DockerLogin(imageTag, loginConfig)
```
Comment (AlexeiVainshtein) [inline artifactory/commands/docker/pull.go]: I think we can remove the imageTag variable and just use the dpc.ImageTag() here. So the `	imageTag := dpc.ImageTag()` will be removed and instead as in the push command all will be done on the dpc.ImageTag()

File: artifactory/commands/docker/push.go [hunk 4]
```diff
@@ -33,18 +33,20 @@ func (dpc *DockerPushCommand) Run() error {
...
 		return err
 	}
-	loginConfig := &docker.DockerLoginConfig{ArtifactoryDetails: rtDetails}
-	imageTag := dpc.ImageTag()
-	err = docker.DockerLogin(imageTag, loginConfig)
-	if err != nil {
-		return err
+	if !dpc.skipLogin {
+		loginConfig := &docker.DockerLoginConfig{ArtifactoryDetails: rtDetails}
+		imageTag := dpc.ImageTag()
```
Comment (AlexeiVainshtein) [inline artifactory/commands/docker/push.go]: I think this can be removed and just to use the dpc.ImageTag()

**Comment by eyalbe4** [comment]
Implemented all comments in https://github.com/jfrog/jfrog-cli-go/pull/388
I'm therefore closing this one.

**Comment by fdolsky321** [comment]
Guys, is there a way to push docker image to artifactory, when the config.json file is restricted to write (by admins). I was trying to use --user and --api-key with --skip-login, but then I was not authorized at all and in artifactory log, I saw HTTP request from anonymous rather then the user which was used in --user. Do you have any idea, is this possible? @eyalbe4 @AlexeiVainshtein 
Thanks a lot.

**Pull request closed without merging**